airbyte-webapp/src/components/ui/PillSelect/PillSelect.tsx

```diff
@@ -22,7 +21,11 @@ export const PillSelect: React.FC<PillSelectProps> = ({ className, ...props }) =
             ? value.map(({ label }: { label: string }) => label).join(", ")
             : value.label
           : "";
-
+        const pillLabel = value
+          ? isMulti
+            ? value.map(({ pillLabel }: { pillLabel: string }) => pillLabel).join(", ")
+            : value.pillLabel
+          : "";
```

Comment:
If the label accepts both an array value and a string, when it's an array we can join with the `" | "` string before joining the values with `, `. Label should be able to handle both cases.